packages/flame/lib/src/game/base_game.dart

```diff
@@ -87,13 +87,34 @@ class BaseGame extends Game with FPSCounter {
     _combinedProjector = Projector.compose([camera, viewport]);
   }
 
+  /// This method setps up the OrderedSet instance used by this game, before
+  /// any lifecycle methods happen.
+  ///
+  /// You can return a specific sub-class of OrderedSet, like
+  /// [QueryableOrderedSet] for example, that we use for Collidables.
+  OrderedSet<Component> createOrderedSet() {
+    final comparator = Comparing.on<Component>((c) => c.priority);
+    if (this is HasCollidables) {
```

Comment:
I am still considering going full QOS always. that would make it easier for people to just register/query w/o casting. For now I thought doing this as it's pretty simple and works as almost an A/B test, and we can see if there is any issues with Collidables by using this feature. we then could consider moving to 100% and changing the typedef of the field.